lib/render/RayCaster.cpp

```diff
@@ -42,23 +45,36 @@ RayCaster::RayCaster( const ParamsMgr*    pm,
                       paramsType,
                       classType,
                       instName,
-                      dataMgr )
+                      dataMgr ),
+            _backFaceTexOffset     ( 0 ),
+            _frontFaceTexOffset    ( 1 ),
+            _volumeTexOffset       ( 2 ),
+            _colorMapTexOffset     ( 3 ),
+            _missingValueTexOffset ( 4 ),
+            _xyCoordsTexOffset     ( 5 ),
+            _zCoordsTexOffset      ( 6 )
 {
     _backFaceTextureId           = 0;
     _frontFaceTextureId          = 0;
     _volumeTextureId             = 0;
     _missingValueTextureId       = 0;
     _colorMapTextureId           = 0;
+    _xyCoordsTextureId           = 0;
+    _zCoordsTextureId            = 0;
     _frameBufferId               = 0;
-    _depthBufferId               = 0;
 
     _vertexArrayId               = 0;
     _vertexBufferId              = 0;
     _indexBufferId               = 0;
+    _vertexAttribId              = 0;
+    _xyCoordsBufferId            = 0;
+    _zCoordsBufferId             = 0;
 
     _1stPassShaderId             = 0;
     _2ndPassShaderId             = 0;
     _3rdPassShaderId             = 0;
+    _3rdPassMode1ShaderId        = 0;
+    _3rdPassMode2ShaderId        = 0;
 
     _drawBuffers[0]              = GL_COLOR_ATTACHMENT0;
     _drawBuffers[1]              = GL_COLOR_ATTACHMENT1;
```

Comment:
OpenGL call should not be made in the constructor:

    GLint viewport[4];
     glGetIntegerv( GL_VIEWPORT, viewport );
     std::memcpy( _currentViewport, viewport, 4 * sizeof(GLint) );

There is no OpenGL context until after RayCaster::_initializeGL is called